test/hotspot/jtreg/serviceability/jvmti/events/Breakpoint/breakpoint01/libbreakpoint01.cpp

```diff
@@ -0,0 +1,321 @@
+    {"bpMethod2V", "()I"}
+};
+
+static const jboolean METHODS_ATTRS[METH_NUM] = {JNI_FALSE, JNI_FALSE, JNI_TRUE, JNI_TRUE};
+
+
+static const char *CLASS_SIG = "Lbreakpoint01;";
+
+static const char *THREAD_NAME = "breakpoint01Thr";
+
+static volatile int bpEvents[METH_NUM];
+static volatile jint result = PASSED;
+static jvmtiEnv *jvmti = NULL;
+static jvmtiEventCallbacks callbacks;
+
+static volatile int callbacksEnabled = NSK_TRUE;
+static jrawMonitorID agent_lock;
+
+static void initCounters() {
+  int i;
+
+  for (i = 0; i < METH_NUM; i++)
+    bpEvents[i] = 0;
+}
+
+static void setBP(jvmtiEnv *jvmti, JNIEnv *jni, jclass klass) {
+  jmethodID mid;
+  jvmtiError err;
+  int i;
+
+  for (i = 0; i < METH_NUM; i++) {
+    mid = jni->GetMethodID(klass, METHODS[i][0], METHODS[i][1]);
+    if (mid == nullptr) {
+      jni->FatalError("failed to get ID for the java method\n");
+    }
+
+    err = jvmti->SetBreakpoint(mid, 0);
+    if (err != JVMTI_ERROR_NONE) {
+      jni->FatalError("failed to set breakpoint\n");
+    }
+  }
+}
+
+/** callback functions **/
+void JNICALL
+ClassLoad(jvmtiEnv *jvmti, JNIEnv *jni, jthread thread, jclass klass) {
+  char *sig, *generic;
+  jvmtiError err;
+
+  RawMonitorLocker rml(jvmti, jni, agent_lock);
+  if (callbacksEnabled) {
+    // GetClassSignature may be called only during the start or the live phase
+    err = jvmti->GetClassSignature(klass, &sig, &generic);
+    if (err != JVMTI_ERROR_NONE) {
+      jni->FatalError("failed to obtain a class signature\n");
+    }
+
+    if (sig != NULL && (strcmp(sig, CLASS_SIG) == 0)) {
+      LOG("ClassLoad event received for the class %s setting breakpoints ...\n", sig);
+      setBP(jvmti, jni, klass);
+    }
+  }
+}
+
+void JNICALL
+Breakpoint(jvmtiEnv *jvmti, JNIEnv *jni, jthread thread, jmethodID method, jlocation location) {
+  jclass klass;
+  char *clsSig, *generic, *methNam, *methSig;
+  jvmtiThreadInfo thr_info;
+  jvmtiError err;
+  int checkStatus = PASSED;
+  int i;
+
+  LOG(">>>> Breakpoint event received\n");
+
+/* checking thread info */
+  err = jvmti->GetThreadInfo(thread, &thr_info);
+  if (err != JVMTI_ERROR_NONE) {
+    result = STATUS_FAILED;
+    LOG("TEST FAILED: unable to get thread info during Breakpoint callback\n\n");
+    return;
+  }
+
+  if (thr_info.name == NULL ||
+      strcmp(thr_info.name, THREAD_NAME) != 0) {
+    result = checkStatus = STATUS_FAILED;
+    LOG(
+        "TEST FAILED: Breakpoint event with unexpected thread info:\n"
+        "\tname: \"%s\"\ttype: %s %s thread\n\n",
+        (thr_info.name == NULL) ? "NULL" : thr_info.name,
+        (jni->IsVirtualThread(thread) == JNI_TRUE) ? "virtual" : "kernel",
+        (thr_info.is_daemon == JNI_TRUE) ? "deamon" : "user");
+  } else {
+    LOG("CHECK PASSED: thread name: \"%s\"\ttype: %s %s thread\n",
+           thr_info.name,        (jni->IsVirtualThread(thread) == JNI_TRUE) ? "virtual" : "kernel",
+        (thr_info.is_daemon == JNI_TRUE) ? "deamon" : "user");
+  }
```

Comment:
I'd suggest to add locals (their names are up to you):
   const char* thr_virtual_tag = jni->IsVirtualThread(thread) == JNI_TRUE ? "virtual" : "kernel";
   const char* thr_daemon_tag == JNI_TRUE) ? "deamon" : "user";

It is better to place togeter lines: 129+130.
Line 137 is not aligned, and there are many unneeded spaces.
The '()' around conditions are not needed. At least, I do not see them increasing readability.

Reply:
fixed

Reply:
Thanks, I'll mark all the comments related to the JVMTI event tests as resolved.